CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/resourcetypes/CVKImage.java

```diff
@@ -959,17 +942,63 @@ public int SaveToFile(File file) {
             ret = CopyToCPUMemoryImage(stack);
             if (VkFailed(ret)) { return ret; }
                         
-            // Start mapping the destination images memory
-            StartMemoryMap(stack);
+            // Start mapping the destination image's memory
+            destImage.StartMemoryMap(stack);
             
             // Save to the given file
-            ret = SaveMemoryToFile(stack, file, false);
+            ret = destImage.SaveMemoryToFile(stack, file);
             if (VkFailed(ret)) { return ret; }
 
             // Cleanup
-            EndMemoryMap();
+            destImage.EndMemoryMap();
+            destImage.Destroy();
+            destImage = null;
      
         }
         return ret;
     }
+    
+    
+    public int ReadPixel(int x, int y) {
+        int r;
+        
+        try (MemoryStack stack = stackPush()) {
+            StartMemoryMap(stack);                 
+            r = ReadPixel(stack, x, y);
+            EndMemoryMap();
+        }
+        
+        return r;
+    }
+    
+    private int ReadPixel(MemoryStack stack, int x, int y) {
+        CVKAssertNotNull(pWriteMemory);
+        
+        int redPixel = 0;
+
+        // Get layout of the image
+        VkImageSubresource subResource = VkImageSubresource.callocStack(stack);
+        subResource.set(VK_IMAGE_ASPECT_COLOR_BIT, 0, 0);
+        VkSubresourceLayout subResourceLayout = VkSubresourceLayout.callocStack(stack);
+        vkGetImageSubresourceLayout(CVKDevice.GetVkDevice(), GetImageHandle(), subResource, subResourceLayout);
+
+        // VK_FORMAT_R32_SFLOAT
+        // Size (in bytes) of the destination image including any padding
+        final long sourceImageSize = subResourceLayout.size(); 
+        // Calculate the file width including the padding (in bytes)
+        int fileWidth = (int)(subResourceLayout.rowPitch() / 4);
+
+        try {
+            FloatBuffer sourceBuffer = pWriteMemory.getFloatBuffer((int)sourceImageSize);
+            redPixel = (int)sourceBuffer.get(x + (y * fileWidth));
+
+        }
+        catch (Exception e) {
+            System.err.println("Error: " + e.getMessage());
+            GetLogger().LogException(e, String.format("Get ID from Image Memory failed"));
```

Comment:
Don't need the string format, LogException has a varargs version.

Reply:
removed